u8timeseries/backtesting/backtesting.py

```diff
@@ -259,3 +260,193 @@ def plot_residuals_analysis(residuals: TimeSeries, num_bins: int = 20):
+    iterator = _build_tqdm_iterator(params_cross_product, verbose)
+    for param_combination in iterator:
+        param_combination_dict = dict(list(zip(parameters.keys(), param_combination)))
+        model = model_class(**param_combination_dict)
+        if (val_series is None):  # expanding window mode
+            backtest_forecast = backtest_forecasting(train_series, model, backtest_start_time, fcast_horizon_n)
+            error = metric(backtest_forecast, train_series)
+        else:  # split mode
+            model.fit(train_series)
+            error = metric(model.predict(len(val_series)), val_series)
+        if (error < min_error):
+            min_error = error
+            best_param_combination = param_combination_dict
+    logger.info('Chosen parameters: ' + str(best_param_combination))
+    return model_class(**best_param_combination)
+
+
+def explore_models(train_series: TimeSeries,
+                   val_series: TimeSeries,
+                   test_series: TimeSeries,
+                   width: int = 3,
+                   metric: Callable[[TimeSeries, TimeSeries], float] = metrics.mape,
+                   verbose: bool = True):
+    """ A function for exploring the suitability of multiple models on a given train/validation/test split.
+
+    This funtion iterates through a list of models, training each on `train_series` and `val_series`
+    and evaluating them on `test_series`. Models with free hyperparameters are first
+    tuned by trying out a 'reasonable' set of hyperparameter combinations. A model for
+    each combination is trained on `train_series` and evaluated on `val_series`.
+    The best-performing combination of hyperparameters (the one resulting in the lowest output of the `metric`
+    function) is then chosen for that type of model. This way, none of the models have 'looked at' the test set
+    before the final evaluation.
+
+    The performance of every model type is then plotted against the series of actual observations.
+    Additionally, the `metric` values and the runtimes of every model type are plotted in bar charts.
+
+    Parameters
+    ----------
+    train_series
+        A TimeSeries instance used for training during model tuning and evaluation.
+    val_series
+        A TimeSeries instance used for validation during model tuning and for training during evaluation.
+    test_series
+        A TimeSeries instance used for validation when evaluating a model.
+    width
+        An integer indicating the number of plots that are displayed in one row.
+    metric:
+        A function that takes two TimeSeries instances as inputs and returns a float error value.
+    verbose:
+        Whether to print progress.
+    """
+
+    raise_if_not(width > 1, "Please choose an integer 'width' value larger than 1", logger)
+
+    # list of tuples containing model classes and hyperparameter selection, if required
+    model_parameter_tuples = [
```

Comment:
How about having this as a default, but let users specify their own `model_parameters_tuples` if they want to?